notebooks/02-ingestion/15-native-dimensions.ipynb

```diff
@@ -0,0 +1,786 @@
+    "\n",
+    "There are two flavors of dimension object:\n",
+    "\n",
+    "* Dimensions that ingested using all defaults, bringing in data as a string with a bitmap index.\n",
+    "* Dimensions that have specific types."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1987c712-8670-4fbd-b3c6-d072efb439a8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dataSchema_dimensionsSpec = {\n",
+    "    \"dimensions\": [\n",
+    "        \"username\",\n",
+    "        \"post_title\",\n",
+    "        {\n",
+    "            \"name\" : \"views\",\n",
+    "            \"type\" : \"long\" },\n",
+    "        {\n",
+    "            \"name\" : \"upvotes\",\n",
+    "            \"type\" : \"long\" },\n",
+    "        {\n",
+    "            \"name\" : \"comments\",\n",
+    "            \"type\" : \"long\" }\n",
+    "        ]\n",
+    "      }"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "65e24a56-3d9e-4589-8d0c-113a5b5ec8e9",
+   "metadata": {},
+   "source": [
+    "Run the next cell to create the final `dataSchema`. Notice that the table name is also defined here.\n",
+    "\n",
+    "Beneath this it is combined with the `ioConfig` and `tuningConfig` to create a native [ingestion specification](https://druid.apache.org/docs/latest/ingestion/ingestion-spec)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c69af4ad-00e2-4d9b-b7fc-5725dfe9e040",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "table_name = topic_name\n",
+    "\n",
+    "dataSchema = {\n",
+    "      \"dataSource\": table_name,\n",
+    "      \"timestampSpec\": dataSchema_timestampSpec,\n",
+    "      \"dimensionsSpec\": dataSchema_dimensionsSpec,\n",
+    "      \"granularitySpec\": dataSchema_granularitySpec\n",
+    "    }\n",
+    "\n",
+    "ingestionSpec = {\n",
+    "  \"type\": \"kafka\",\n",
+    "  \"spec\": {\n",
+    "    \"ioConfig\": ioConfig,\n",
+    "    \"tuningConfig\": tuningConfig,\n",
+    "    \"dataSchema\": dataSchema\n",
+    "  }\n",
+    "}\n",
+    "\n",
+    "print(json.dumps(ingestionSpec, indent=5))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fe50d1bb-1bc9-4ef1-a1a7-637cc3bb4616",
+   "metadata": {},
+   "source": [
+    "Start the ingestion of the raw data from Apache Kafka by submitting this object to Apache Druid by running the cell below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "92aa5125-1096-462a-a637-cd8f438a2074",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "supervisor = requests.post(f\"{druid_host}/druid/indexer/v1/supervisor\", json.dumps(ingestionSpec), headers=generalHeaders)\n",
+    "print(supervisor.status_code)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "dc897578-48d7-4b8c-9ece-dab4390e2336",
+   "metadata": {},
+   "source": [
+    "Run the following cell to wait until the ingestion has started and the new table is ready for query."
```

Comment:
"Run the cell to wait" sounds a bit odd.

Reply:
Yeah this was too lengthy... I've merged it into a single cell.